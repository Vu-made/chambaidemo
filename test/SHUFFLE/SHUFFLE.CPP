#include <bits/stdc++.h>
#define N 100005
using namespace std;
int Next[N], Pre[N];
int n,x;
int main()
{
    freopen("SHUFFLE.INP","r",stdin);
    freopen("SHUFFLE.OUT","w",stdout);
    cin >> n >> x;
    Next[0] = 1; Pre[n+1] = n;
    for (int i=1;i<=n; i++)
    {
        Next[i] = i+1;
        Pre[i] = i-1;
    }
    for (int i,j,k=1;k <= x;k++)
    {
        cin >> i >> j;
        Next[Pre[i]] = Next[i];
        Pre[Next[i]] = Pre[i];
        Next[i] = j;
        Pre[i] = Pre[j];
        Next[Pre[j]] = i;
        Pre[j] = i;
    }
    for (int p=Next[0]; p<n+1; p=Next[p])
        cout << p << " ";
    return 0;
}
