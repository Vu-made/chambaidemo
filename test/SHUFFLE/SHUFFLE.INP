9 3
8 2
4 7
1 9
