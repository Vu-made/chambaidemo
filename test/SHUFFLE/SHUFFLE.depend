# depslib dependency file v1.0
1602671316 source:c:\users\admin\desktop\bgdv\bt\thuat_toan\array1d_3\shuffle\shuffle.cpp
	<bits/stdc++.h>

