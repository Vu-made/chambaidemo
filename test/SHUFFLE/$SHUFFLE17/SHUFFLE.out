107 55 97 134 14 138 6 35 88 133 123 104 32 45 119 96 132 25 127 90 4 114 130 28 71 3 106 124 5 23 80 92 43 81 65 38 36 51 31 49 72 9 61 34 40 64 94 125 78 75 126 113 98 121 69 117 50 27 15 22 108 99 10 120 102 74 128 52 57 129 139 21 67 70 87 85 136 39 79 89 116 131 135 86 122 103 1 109 41 118 66 20 2 46 59 82 8 44 58 19 24 84 93 54 18 62 17 33 105 137 11 47 30 95 111 7 100 91 101 13 112 42 115 63 110 68 73 83 26 29 37 76 53 12 60 16 48 56 77 